touchforms/formplayer/static/formplayer/script/entrycontrols_full.js

```diff
@@ -201,7 +201,7 @@ function PhoneEntry(question, options) {
 
     this.getErrorMessage = function(rawAnswer) {
         if (rawAnswer === '') { return null; }
-        return (!(/^\+?[0-9]+$/.test(rawAnswer)) ? "This does not appear to be a valid phone/numeric number" : null);
+        return (!(/^\+?-?[0-9.]+$/.test(rawAnswer)) ? "This does not appear to be a valid phone/numeric number" : null);
```

Comment:
It's a little weird that this allows values like `-....3` and `+-3` and `3.`

How about something like `/^[+\-]?\d*(\.\d+)?$/`?